pkg/policy/l4.go

```diff
@@ -43,6 +44,88 @@ type L4Filter struct {
 	Ingress bool
 }
 
+// FillPortRuleHTTP fills in the http port rule fields
+// as a regex into a single string
+func FillPortRuleHTTP(httpRule []api.PortRuleHTTP) []AuxRule {
+	l7rules := []AuxRule{}
+	for _, h := range httpRule {
+		r := AuxRule{}
+
+		if h.Path != "" {
+			r.Expr = "PathRegexp(\"" + h.Path + "\")"
+		}
+
+		if h.Method != "" {
+			if r.Expr != "" {
+				r.Expr += " && "
+			}
+			r.Expr += "MethodRegexp(\"" + h.Method + "\")"
+		}
+
+		if h.Host != "" {
+			if r.Expr != "" {
+				r.Expr += " && "
+			}
+			r.Expr += "HostRegexp(\"" + h.Host + "\")"
+		}
+
+		for _, hdr := range h.Headers {
+			s := strings.SplitN(hdr, " ", 2)
+			if r.Expr != "" {
+				r.Expr += " && "
+			}
+			r.Expr += "Header(\""
+			if len(s) == 2 {
+				// Remove ':' in "X-Key: true"
+				key := strings.TrimRight(s[0], ":")
+				r.Expr += key + "\",\"" + s[1]
+			} else {
+				r.Expr += s[0]
+			}
+			r.Expr += "\")"
+		}
+
+		if r.Expr != "" {
+			l7rules = append(l7rules, r)
+		}
+	}
+
+	return l7rules
+}
+
+// FillPortRuleKafka fills in the Kafka port rule fields
+// as a regex into a single string
+func FillPortRuleKafka(kafkaRule []api.PortRuleKafka) []AuxRule {
+	l7rules := []AuxRule{}
+	for _, h := range kafkaRule {
+		r := AuxRule{}
+		r.L7type = "kafka"
+
+		if h.ApiVersion != "" {
+			r.Expr = "ApiVersionRegexp(\"" + h.ApiVersion + "\")"
```

Comment:
You didn't rename the usages of ApiVersion and ApiKey when you renamed them in rule.go. I'd recommend using automatic refactoring tools (e.g. Gogland) instead of doing such refactorings by hand.

Reply:
done